app/components/FileLoader.js

```diff
@@ -41,6 +42,24 @@ export default class FileLoader extends Component {
     topNotifOffset: PropTypes.number.isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      files: [],
+      selections: [],
+    };
+  }
+
+  static getDerivedStateFromProps(newProps, oldState) {
+    if (newProps.store.files !== oldState.files) {
+      return {
+        files: newProps.store.files,
+        selections: [],
+      };
+    }
+    return oldState;
```

Comment:
According to the docs you can return `null` to not change state. Also in general I don't love managing state like this. I think putting selections in redux might actually be better/more flexible. But for now this is fine.